src/util/schnorr.rs

```diff
@@ -49,11 +49,7 @@ pub trait TapTweak {
 impl TapTweak for UntweakedPublicKey {
     fn tap_tweak<C: Verification>(&self, secp: Secp256k1<C>, merkle_root: Option<TapBranchHash>) -> TweakedPublicKey {
         // Compute the tweak
-        let mut engine = TapTweakHash::engine();
-        engine.input(&self.serialize());
-        merkle_root.map(|hash| engine.input(&hash));
-        let tweak_value: [u8; 32] = TapTweakHash::from_engine(engine).into_inner();
-            
+        let tweak_value = TapTweakHash::from_key_and_tweak(self.clone(), merkle_root).into_inner();
```

Comment:
Isn't `clone()` expensive?

Reply:
schnorr::PublicKey is Copy, I think it is the same.

Reply:
In that case `.clone()` is redundant and I don't think it should be used.

Reply:
`TapTweakHash::from_key_and_tweak()` takes in `schnorrsig::PublicKey` where as the function it is being called in has `&UntweakedPublicKey`. 

I used the `.clone()` to get around the borrow checker. Should I instead pass in `*self` to the method or should I modify `TapTweakHash::from_key_and_tweak()` to take in `&schnorrsig::PublicKey` as a parameter?

Reply:
I think taking `self` by value everywhere is the correct approach. The compiler should insert references for optimization if needed.

Reply:
Fixed in the latest commit and squashed.